### PR TITLE
Picture improvements

### PR DESCRIPTION
**Pull request by garbear**
As discussed at devcon, we really want to get as many people as possible involved in improving the quality and architecture of our code. New features are awesome, but the legacy code from our early days is making it hard to properly implement these features. The pictures code is one of the best examples of an area with lots of low-hanging fruit, so I figured I'd start there with some general cleanup.

### CODE REVIEW

File: xbmc/pictures/GUIWindowPictures.cpp [hunk 1]
```diff
@@ -602,9 +602,11 @@ void CGUIWindowPictures::OnItemInfo(int itemNumber)
 
 std::string CGUIWindowPictures::GetStartFolder(const std::string &dir)
 {
-  std::string lower(dir); StringUtils::ToLower(lower);
-  if (lower == "plugins" || lower == "addons")
+  if (StringUtils::EqualsNoCase(dir, "plugins") ||
```
Comment (xhaggi) [inline xbmc/pictures/GUIWindowPictures.cpp]: you've changed the implementation from equals to equals no case and not only the readability ;)

Comment (xhaggi) [inline xbmc/pictures/GUIWindowPictures.cpp]: not sure if we really want it to be changed

Comment (garbear) [inline xbmc/pictures/GUIWindowPictures.cpp]: Doesn't the `StringUtils::ToLower(lower)` have the effect of a case-insensitive comparison? So this isn't a change in functionality, is it?

Comment (xhaggi) [inline xbmc/pictures/GUIWindowPictures.cpp]: ohh sorry for the noise, only saw the conditions changed 😀

File: xbmc/pictures/GUIWindowPictures.cpp [hunk 2]
```diff
@@ -607,9 +602,11 @@ void CGUIWindowPictures::OnItemInfo(int itemNumber)
...
+      StringUtils::EqualsNoCase(dir, "addons"))
+  {
```
Comment (Montellese) [inline xbmc/pictures/GUIWindowPictures.cpp]: Why the curly braces? That doesn't fit our coding style.

File: xbmc/windows/GUIMediaWindow.cpp [hunk 3]
```diff
@@ -1990,9 +1990,12 @@ bool CGUIMediaWindow::Filter(bool advanced /* = true */)
 
 std::string CGUIMediaWindow::GetStartFolder(const std::string &dir)
 {
-  std::string lower(dir); StringUtils::ToLower(lower);
-  if (lower == "$root" || lower == "root")
+  if (StringUtils::EqualsNoCase(dir, "$root") ||
+      StringUtils::EqualsNoCase(dir, "root"))
+  {
```
Comment (Montellese) [inline xbmc/windows/GUIMediaWindow.cpp]: Same as above with curly braces.

**Review by Montellese — CHANGES_REQUESTED** [review]
Looks good apart from the cosmetics.

File: xbmc/pictures/GUIViewStatePictures.cpp [hunk 4]
```diff
@@ -82,8 +82,20 @@ std::string CGUIViewStateWindowPictures::GetExtensions()
 VECSOURCES& CGUIViewStateWindowPictures::GetSources()
 {
   VECSOURCES *pictureSources = CMediaSourceSettings::GetInstance().GetSources("pictures");
+
+  // Guard against source type not existing
+  if (pictureSources == nullptr)
```
Comment (Paxxi) [inline xbmc/pictures/GUIViewStatePictures.cpp]: You do not need the explicit nullptr check, `if(pictureSources) ` is fine

**Review by Paxxi — CHANGES_REQUESTED** [review]
Add the braces montellese asks for and it should be good

File: xbmc/pictures/GUIViewStatePictures.cpp (hunk 4 above)
Comment (garbear) [inline xbmc/pictures/GUIViewStatePictures.cpp]: i'm an explicit programmer

Comment (Paxxi) [inline xbmc/pictures/GUIViewStatePictures.cpp]: Yup, both are fine, just a matter of taste

**Comment by garbear** [comment]
braces removed. jenkins build this please

**Pull request merged**